instrumentation/nocode/src/main/java/com/splunk/opentelemetry/instrumentation/nocode/YamlParser.java

```diff
@@ -19,93 +19,233 @@
+    parsers.addAll(Arrays.asList(extras));
+    for (MatcherParser mp : parsers) {
+      answer.put(mp.name(), mp);
+    }
+    return answer;
+  }
+
+  private static List<ElementMatcher> matcherListFromYaml(
+      Object yamlObject, Map<String, MatcherParser> parsers) {
+    if (!(yamlObject instanceof List)) {
+      throw new IllegalArgumentException("Yaml list value expected: " + yamlObject);
+    }
+    List<Object> yaml = (List<Object>) yamlObject;
+    ArrayList<ElementMatcher> answer = new ArrayList<>();
+    for (Object sub : yaml) {
+      answer.add(matcherFromYaml(sub, parsers));
+    }
+    return answer;
+  }
+
+  private static ElementMatcher matcherFromYaml(
+      Object yamlObject, Map<String, MatcherParser> parsers) {
+    if (!(yamlObject instanceof Map)) {
+      throw new IllegalArgumentException("Bare yaml value not expected: " + yamlObject);
+    }
+    Map<String, Object> yaml = (Map<String, Object>) yamlObject;
+    if (yaml.size() != 1) {
+      throw new IllegalArgumentException("Multiple yaml elements not allowed without and:/or:");
+    }
+    String key = yaml.keySet().iterator().next();
+    Object value = yaml.get(key);
+    return matcherFromKeyAndYamlValue(key, value, parsers);
+  }
+
+  private static ElementMatcher matcherFromKeyAndYamlValue(
+      String key, Object value, Map<String, MatcherParser> parsers) {
+    MatcherParser parser = parsers.get(key);
+    if (parser == null) {
+      throw new IllegalArgumentException("Unknown yaml element: " + key);
+    }
+    return parser.parse(value, parsers);
+  }
+
+  enum MatcherParser {
+    not {
+      ElementMatcher parse(Object value, Map<String, MatcherParser> parsers) {
+        return ElementMatchers.not(matcherFromYaml(value, parsers));
+      }
+    },
+    and {
+      ElementMatcher parse(Object value, Map<String, MatcherParser> parsers) {
+        ElementMatcher.Junction matcher = ElementMatchers.any();
+        for (ElementMatcher sub : matcherListFromYaml(value, parsers)) {
+          matcher = matcher.and(sub);
+        }
+        return matcher;
+      }
+    },
+    or {
+      ElementMatcher parse(Object value, Map<String, MatcherParser> parsers) {
+        ElementMatcher.Junction matcher = ElementMatchers.none();
+        for (ElementMatcher sub : matcherListFromYaml(value, parsers)) {
+          matcher = matcher.or(sub);
+        }
+        return matcher;
+      }
+    },
+    named {
+      ElementMatcher parse(Object value, Map<String, MatcherParser> parsers) {
+        return ElementMatchers.named(value.toString());
+      }
+    },
+    nameMatches {
+      ElementMatcher parse(Object value, Map<String, MatcherParser> parsers) {
+        return ElementMatchers.nameMatches(value.toString());
+      }
+    },
+    hasSuperType {
+      ElementMatcher parse(Object value, Map<String, MatcherParser> parsers) {
+        return ElementMatchers.hasSuperType(ElementMatchers.named(value.toString()));
```

Comment:
we usually use `hasSuperType` from `io.opentelemetry.javaagent.extension.matcher.AgentElementMatchers`. I think it behaves better when super type of the tested type can't be found